### PR TITLE
Combine All the IAM cypress project update tests

### LINKED ISSUE #2299: Combine All the IAM cypress project update tests
## User Story

There are Client Runs, Actions, Compliance, and Nodemanager tests for project update re-tagging. This task is to combine all 4 of them to have only one "apply-rules" request. This will decrease the time it takes to run the cypress tests in buildkite. 

Below are the tests to combine:
* https://github.com/chef/automate/blob/master/e2e/cypress/integration/api/iam/06_projects_api_compliance_update.ts
* https://github.com/chef/automate/blob/master/e2e/cypress/integration/api/iam/07_projects_api_client_runs_update.ts
* https://github.com/chef/automate/pull/2280
* #2192

## Definition of Done
Only one "apply-rules" request is made for the Client Runs, Actions, Compliance, and Nodemanager tests project update re-tagging tests. 

## Demo Script / Repro Steps
Run the tests in buildkite and ensure the tests run faster.

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
Combine all the ingestion project tagging integration tests to minimize the time the tests take to run. This will eliminate the apply-rules from being called 4 times which takes 2 minutes to run. 

### :chains: Related Resources
fixes #2299 

### :athletic_shoe: How to Build and Test the Change
Check the buildkite tests

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts [hunk 1]
```diff
@@ -217,67 +217,92 @@ describeIfIAMV2p1('Nodemanager ingestion project tagging', () => {
     });
 
     // wait for the report to be ingested
-    waitForNodemanagerNode(complianceNodeId, 30);
+    cy.waitForNodemanagerNode(complianceNodeId, 30);
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts]: could have just one comment above `wait for all nodes to be ingested`

File: e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts [hunk 2]
```diff
@@ -217,67 +217,92 @@ describeIfIAMV2p1('Nodemanager ingestion project tagging', () => {
...
-          per_page: 100
-        }
-      }).then((response) => {
-        expect(response.body.nodes.length).to.be.greaterThan(0);
-        expect(response.body.nodes.some((node: any) =>
-          node.id === complianceNodeId)).to.equal(true);
-        expect(response.body.nodes.some((node: any) =>
-          node.id === clientRunsNodeId)).to.equal(true);
+  describe('Node Manager', () => {
+    // Ensure that both the compliance and client run nodes' tags are updated
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts]: maybe add to this: `by fetching the nodes with the expected project filter` (and repeat for the similar comments in the rest of the file)

File: e2e/cypress/support/commands.ts [hunk 3]
```diff
@@ -347,3 +316,60 @@ function waitUntilAdminTokenPermissioned(attempts: number): void {
   });
   }
 }
+
+Cypress.Commands.add('waitForClientRunsNode', (nodeId: string, maxRetries: number) => {
```
Comment (bcmdarroch) [inline e2e/cypress/support/commands.ts]: can you move these above the helpers section (line 307)?

File: e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts [hunk 4]
```diff
@@ -217,67 +217,92 @@ describeIfIAMV2p1('Nodemanager ingestion project tagging', () => {
     });
 
     // wait for the report to be ingested
-    waitForNodemanagerNode(complianceNodeId, 30);
+    cy.waitForNodemanagerNode(complianceNodeId, 30);
+    cy.waitForComplianceNode(complianceNodeId, start, end, 30);
     // wait for the client run report to be ingested
-    waitForNodemanagerNode(clientRunsNodeId, 30);
+    cy.waitForNodemanagerNode(clientRunsNodeId, 30);
+    cy.waitForClientRunsNode(clientRunsNodeId, 30);
```
Comment (bcmdarroch) [inline e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts]: what are L223 and L224 doing differently? they look like they're waiting on the same client runs node.

Comment (lancewf) [reply e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts]: Line 223 is waiting for the node manager to ingest the node and line 224 is waiting for Client Runs ingestion. Before we were just waiting on the node manager it turns out that node manager is faster than Client Runs.

File: e2e/cypress/support/commands.ts (hunk 3 above)
Comment (lancewf) [reply e2e/cypress/support/commands.ts]: Thanks.

**Pull request merged**